### ISSUE TITLE
✅ 🐞 Bug Report: Video widget appears half of the height when visited using Safari after 3.11.1 - [ED-10100]

### ISSUE THREAD

**Issue by ada2023** [elementor/elementor#21454]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Video widget heights are not shown correctly mainly in Safari, on some websites also in other browsers, after updating to 3.11.1. The new height is about 50% of initial height of embedded YouTube / Vimeo videos in some browsers, mainly Safari, compromising usability and design. On some websites, the same videos still show correctly in Chrome, Edge, Vivaldi, Opera at the same time. On other websites the same issue is also seen in Firefox and Chrome.

### Steps to reproduce

1. Open a page with Elementor version 3.11.1/3.11.2/3.11.3 containing an Elementor video widget (YouTube, Vimeo) in Safari.
2. Compare with views in other browsers (Chrome, Firefox, Vivaldi, ...)
3. Compare with Elementor version 3.10.2

### Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

Server Environment |   |  
-- | -- | --
Operating System: | Linux |  
Software: | Apache/2.4.41 (Ubuntu) OpenSSL/1.1.1f mod_fcgid/2.3.9 |  
MySQL version: | (Ubuntu) v5.7.39 |  
PHP Version: | 8.0.28 |  
PHP Memory Limit: | 128M |  
PHP Max Input Vars: | 1000 |  
PHP Max Post Size: | 20M |  
GD Installed: | Yes |  
ZIP Installed: | Yes |  
Write Permissions: | All right |  
Elementor Library: | Connected

**Comment by nicholaszein** [comment]
Hello @ada2023, 

Thank you for submitting your report! 🙏 

### ❌ While reviewing your report, we noticed it was missing some key pieces of information:


- 💾 **System info:** Please include the System Info report as outlined in our [guidelines](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml). 
   - Get your System Info from your WordPress `Dashboard` > `Elementor` > `System info` [learn how to copy it here](https://elementor.com/help/hare-my-system-info/). We can't accept issues without it.

- 🗑️ **Clear the cache:** Always clear the cache after any update (WordPress, Theme, Plugins, Posts, and Pages). This includes but is not limited to server cache, site cache, CDN cache, and browser cache.

- 💻**Browser Version:** Please share the version of the browser you're experiencing the issue. 

Without them, we simply can't help. 😕 

### 🧐 Please provide the missing information above so we can troubleshoot your issue faster and better.


Best regards

**Comment by ada2023** [comment]
Hi Nicholas,

thanks for your answer. 
Although system info is not needed to inspect this issue, pls find it below. Pls also ask your team to accept doing without it in cases where it is not needed as it often contains sensitive customer data (that we are not authorized to share under GDPR, as several other people have already mentioned)  and always asking for it just slows down bug fixing. 
Caches were all cleared.
Browser: Safari 14.
We are not the only ones experiencing this issue starting with 3.11.1 (I have seen 3 support forum questions regarding this so far), so please do look into it. Looking forward to hearing from you.

Best regards


== Server Environment ==
	Operating System: Linux
	Software: Apache/2.4.41 (Ubuntu) OpenSSL/1.1.1f mod_fcgid/2.3.9
	MySQL version: (Ubuntu) v5.7.39
	PHP Version: 8.0.28
	PHP Memory Limit: 128M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 20M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.1.1
	Site URL: REDACTED
	Home URL: REDACTED
	WP Multisite: No
	Max Upload Size: 20 MB
	Memory limit: 256M
	Max Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: de-DE
	Timezone: Europe/Vienna
	Admin Email: REDACTED
	Debug Mode: Inactive

== Theme ==
	Name: REDACTED
	Version: 1.0.0.1612185305
	Author: REDACTED
	Child Theme: Yes
	Parent Theme Name: Astra
	Parent Theme Version: 4.0.2
	Parent Theme Author: Brainstorm Force

== User ==
	Role: administrator
	WP Profile lang: de_DE
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36

== Active Plugins ==
	Backup Duplicator & Migration - WP STAGING
		Version: 2.12.0
		Author: WP-STAGING, WPStagingBackup

	Better Search Replace
		Version: 1.4.2
		Author: WP Engine

	Cache Enabler
		Version: 1.8.13
		Author: KeyCDN

	Carousel Slider
		Version: 2.2.0
		Author: Sayful Islam

	Classic Editor
		Version: 1.6.2
		Author: WordPress Contributors

	Coming Soon Page, Maintenance Mode, Landing Pages & WordPress Website Builder by SeedProd
		Version: 6.15.7
		Author: SeedProd

	Complianz | GDPR/CCPA Cookie Consent
		Version: 6.4.1
		Author: Really Simple Plugins

	Disable Comments
		Version: 2.4.3
		Author: WPDeveloper

	Easy Table of Contents
		Version: 2.0.45.1
		Author: Magazine3

	Elementor
		Version: 3.11.2
		Author: Elementor.com

	Flexy Breadcrumb
		Version: 1.2.1
		Author: PressTigers

	HubSpot All-In-One Marketing - Forms, Popups, Live Chat
		Version: 10.0.19
		Author: HubSpot

	Newsletter, SMTP, Email marketing and Subscribe forms by Sendinblue
		Version: 3.1.57
		Author: Sendinblue

	OMGF
		Version: 5.5.5
		Author: Daan from Daan.dev

	Popup Maker
		Version: 1.17.1
		Author: Popup Maker

	Premium Addons for Elementor
		Version: 4.9.49
		Author: Leap13

	Rank Math SEO
		Version: 1.0.107.3
		Author: Rank Math

	Simple Author Box
		Version: 2.50
		Author: WebFactory Ltd

	Site Kit by Google
		Version: 1.94.0
		Author: Google

	Tracking Code Manager
		Version: 2.0.16
		Author: Data443

	UpdraftPlus - Backup/Restore
		Version: 1.22.24
		Author: UpdraftPlus.Com, DavidAnderson

	Wordfence Security
		Version: 7.9.0
		Author: Wordfence

	WP Crontrol
		Version: 1.15.1
		Author: John Blackbourn & crontributors

	WPForms Lite
		Version: 1.8.0.1
		Author: WPForms

	WPScan
		Version: 1.15.6
		Author: WPScan Team

	Yoast Duplicate Post
		Version: 4.5
		Author: Enrico Battocchi & Team Yoast

**Comment by nicholaszein** [comment]
Hi @ada2023!

Thanks for your feedback! 🙏 

I want to reiterate how important is to share the **<ins>FULL</ins> System Info**:

There is information there, like the information you keep out (logs, active experiments, etc) that is **<ins>CRUCIAL</ins>** for the team to be able to correctly reproduce your issue. We ask for the System Info because **<ins>IT IS ALWAYS REQUIRED</ins>**, even when you think it's not.

If you're concerned about privacy, you can very well redact whatever sensitive information you'd like, as long as we can see the entire System Info and use the relevant information to reproduce and validate your issue.

As for other issues related to yours, please list them here so I can check what you are talking about because since 3.11 there have been no issues related to the video widget.

Please provide the missing information. Thank you in advance! 🙏 

Kind regards.

**Comment by ciprianpentelescu** [comment]
its easy, the problem is introduced in the latest version, in frontent.min.css:

.elementor-widget-video .elementor-wrapper iframe,.elementor-widget-video .elementor-wrapper video {
    height: 100%; <- here is the problem, it should be max-height
    width: 100%;
    border: none;
    background-color: #000
}

**Comment by ada2023** [comment]
@ciprianpentelescu @nicholaszein 

Hi Ciprian,

thx for trying to help. Really appreciate that!! Maybe this also helps pointing the Elementor dev team in the right direction ...
Did you also encounter the issue on your Elementor websites after updating to 3.11 and if yes, which browsers were affected?

Yes, obviously the issue was introduced by Elementor 3.11, therefore I still hope they will **1. finally have a look, 2. fix it asap.**

Although your CSS looks good, it does not work in Safari(14) (there might still be issues with position:absolute in Safari, idk). In case the issue can be fixed by your CSS in other browsers (I did not test, but it might help others), I would however like to suggest the following small change to your CSS selector to also include YouTube (iframe is apparently only used for Vimeo):

.elementor-widget-video .elementor-wrapper {
position: relative;
overflow: hidden;
}

.elementor-widget-video .elementor-wrapper iframe, .elementor-widget-video .elementor-wrapper video {
position: absolute;
top: 0;
left: 0;
width: 100%;
height: 100%;
}

PS: As I could not find a way to fix the Safari problem with CSS (at least not without giving the videos a fixed px height which I did not want to do), my fix was rolling back to 3.10.2.

@nicholaszein any news? I am looking forward to seeing the status of this thread changed to "evaluating" ...

**Comment by brandworker** [comment]
We have the same mistake. Is there a bugfix coming here?

**Comment by ada2023** [comment]
@heysash @brandworker @ciprianpentelescu @nicholaszein 
I just installed the new update 3.11.3 that apparently fixes some issue related to image elements in Safari (14), but unfortunately the VIDEO ISSUE PERSISTS ... Very disappointing.

**Comment by nicholaszein** [comment]
@ada2023 this bug report was not part of this release.

Please be patient. I understand your frustration, but the issue you're reporting is not consistently reproduced.

Kind regards

**Comment by brandworker** [comment]
Is there something new?

**Comment by ada2023** [comment]
@brandworker unfortunately, no. Tried 3.11.4, issue still persists.

**Comment by ada2023** [comment]
@nicholaszein looks like Elementor Pro support at least reproduced the bug, I hear, but still no solution in view. I have given up on asking you.

@hein-obox as you have worked on similar issues, maybe you could take a a quick look? This would be much appreciated :)

**Comment by nicholaszein** [comment]
@ada2023

This issue has been resolved in the latest version of Elementor. I just wasn't able to let you know yet.

Please understand that there is a process to be followed, and your issue is not the only one.

**Comment by nicholaszein** [comment]
Following up,

After the latest release of Elementor (3.11.4) I cannot reproduce the issue reported. Please let us know if this was solved or share the URL of the website in question so we can see the issue for ourselves.


https://user-images.githubusercontent.com/113993460/224712928-e9191a17-3262-444c-8414-b348df34c892.mp4



Thanks

**Comment by brandworker** [comment]
Hello we still have the error. In the pictures you can see the video box. In the backend it is displayed correctly. But in the frontend under Safari and Chrome the video appears clearly cropped in height. We have now restored the old backup. 
<img width="1235" alt="Bildschirm­foto 2023-03-13 um 15 00 31" src="https://user-images.githubusercontent.com/17851162/224726521-e09f4a67-2805-4368-b050-e07b434f1b19.png">
<img width="1235" alt="Bildschirm­foto 2023-03-13 um 15 04 15" src="https://user-images.githubusercontent.com/17851162/224726559-5906a7d6-c63d-4d1f-8665-5923b1f2aeb6.png">
<img width="1235" alt="Bildschirm­foto 2023-03-13 um 15 00 53" src="https://user-images.githubusercontent.com/17851162/224726568-38d32696-0706-4d28-a1e0-8a1cd0960063.png">

**Comment by brandworker** [comment]
@nicholaszein do you see the problem?

**Comment by nicholaszein** [comment]
Thanks @brandworker,

can you share the URL to the page in question?

**Comment by brandworker** [comment]
@nicholaszein 
As of we have restored the old version. So at the moment you can no longer see the error. You must see it, right?
Can I send the link privately?

**Comment by nicholaszein** [comment]
@brandworker, our QA engineers need to be able to reproduce the issue.

Since this is not happening, if we can see a link to a page where this issue is happening, maybe we can find the root of it.

Please do share the link here if possible. We don't answer private messages. Thank you very much for your help! We really appreciate it!🙏

**Comment by brandworker** [comment]
Hello I was able to fix the error. This occurred with us in interaction with the plugin Borlabs Cookies. Youtube video are blocked by the plugin and only displayed when the visitor agrees. With this consent, the effect occurs. If we activate the video wrapper in Borlabs Cookie Blocker, the error is gone. 

Maybe this helps someone here in the stream.

Thanks to the Elementor team and @nicholaszein

**Comment by nicholaszein** [comment]
Great finding @brandworker,

this is why it's important to make sure the issue only happens with Elementor and Elementor Pro activated.

@ada2023, @ciprianpentelescu, @heysash, does deactivating the cookie consent plugin fixes the issue for you too?

**Comment by ada2023** [comment]
@nicholaszein no, it does not fix the issue. We do not use Borlabs Cookies.

**Comment by nicholaszein** [comment]
@ada2023 you are using Complianz... 

Please test with Complianz turned off. The faster you collaborate, the fazer we can find the root of the issue and come up with a solution. As I mentioned, we **_cannot_** recreate your issue. So we need your collaboration to solve it.

Thanks.

**Comment by mobilemichie** [comment]
@nicholaszein Do the engineers still need an instance to view? w no cacheing or cookie complience plugins? 
I have a staging copy of site where you can view this issue with the Safari browser. In our case the boss uses Safari 14.1 and you need this older version of browser to recreate. So farasI can tell Safari 15+ recognises aspect ratio.

**Comment by ada2023** [comment]
@mobilemichie thx!!
@nicholaszein I have already tested with Complianz (and all other plugins) off 2 weeks ago (see above) and again yesterday.

As the issue persists (at least) in Safari(14), I have reverted to 3.10.2 again, so nothing to see here. It is incomprehensible why you insist that you cannot recreate the issue: why don't you just open any website containing any Elementor 3.11.xx YT or Vimeo video widget in Safari(14), seems easy enough? I am a user/customer (who provides services to my customers) and not a co-worker, so I would like the Elementor engineering team to do the work in fixing the bug they created. It is disappointing, to say the least, that you seem to view things differently even after several users/customers turned to you with this issue.

By the way, what @brandworker said is not that they deactivated Borlabs Cookies, but that they added / activated a video wrapper in Borlabs Cookies  (an option that is not available in Complianz), which is probably why it worked as there obviously is an issue in the Elementor video widget wrapper CSS that happened while creating new features for 3.11...

I will now wait patiently until a fix is available without any further requests to you as I have no more time for discussions. Kindly let me know in a timely manner as soon as a real fix is available for all browsers. If your engineers (eg @hein-obox) have a technical question I can help with without openly sending links to client websites, I am happy to answer. Wishing you and the team a good and productive day!

**Comment by nicholaszein** [comment]
Hello @mobilemichie,

thank you so much! That would definitely help us! 🙏 

We want to fix this issue, any further information will help!

**Comment by nicholaszein** [comment]
Hello @ada2023...

Please, understand that no one here likes to waste time. You're implying we are not testing and trying to reproduce your issue, when in fact it's all I've been trying to do. I'm trying to find a common denominator here, since simply reproducing your issue on any Safari browser equal to or newer than version 14 has not shown this issue. So there is definitely something else causing this. We just need to find what it is, and it really doesn't hurt you, in fact, will only help expedite things if you try at least a little bit to be cooperative instead of arguing with me, and speculating if I'm lying to you or not.

I am asking for more information in order to help fix an issue that we clearly can't reproduce. You don't need to share the information at all. But understand that choosing to battle with me to win an argument won't make fixing this issue faster.

Let me try to explain why this isn't something that was caused during the v3.11 update:

No changes were made to the widget's CSS in this version. Therefore there is something else contributing to cause this issue. I am here, asking for information so we can figure out what it is. That's it. I'm here to help. But I can only help if you let me, as other users were kind enough to understand. No one here is wanting to dismiss this as not a bug. If that were the case, this issue would have been long closed.

So please, if you can, work with me, I'm just a person trying to help. There is nothing else.

Thanks

PS: Please, don't tag Elementor staff if they are not participating in this threat. Thank you.

**Comment by mobilemichie** [comment]
@nicholaszein Attached here is my system info for the staging copy as of today's update.
The site is staging.ehinstitute.org.

If you would like to see the issue in the wild there are many videos on this page for view in 14.1.2 (Firefox and Chrome both display fine) : https://staging.ehinstitute.org/video-library/

As a note - I found a site that tells you if you can use a CSS property. Aspect-ratio 

I will check in again in late afternoon Pacific to see if you need anything else from. Out of office today so I won't have my computer with me all times.

Cheers,
Michelle
[system-info-staging.ehinstitute.org-14-03-2023-2.txt](https://github.com/elementor/elementor/files/10971964/system-info-staging.ehinstitute.org-14-03-2023-2.txt)

PS. If I get any of the GitHub protocols wrong, please forgive me. newer user. :)

**Comment by nicholaszein** [comment]
Thank you for the feedback @mobilemichie, you are very kind! 🙏 

----

### Hello everyone, we have news. It seems someone managed to reproduce the issue and a fix is underway! 🙌 

Thank you all for your patience! Expect a patch to be released soon.

**Comment by mobilemichie** [comment]
@nicholaszein thank you for the news and thank the engineers for persisting. I really appreciate your time and effort.
:)

**Comment by jc-gno** [comment]
@nicholaszein Has this been fixed? I am still seeing this with all new updates. Thank you!

**Comment by ada2023** [comment]
@jc-gno Should be fixed in upcoming 3.13 update thanks to @rami-elementor. @nicholaszein, can you confirm?

**Comment by rami-elementor** [comment]
I confirm, the fix was merged to 3.13.

**Comment by jc-gno** [comment]
Hello, I have just updated to the newest elementor and the problem still appears. Can anyone else confirm or deny this fix?

**Comment by azanoni** [comment]
For fixing this (an safari too) please add this piece of css

.elementor-widget-video .elementor-wrapper iframe{
min-height:313px;
}

it's not 100% pixel perfect solution but at least we dont need to wait to 3.13 version

**Comment by nicholaszein** [comment]
Hello, everyone!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor **v3.13.0**! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

Cheers 🥂
